core/src/main/java/org/elasticsearch/cluster/routing/UnassignedInfo.java

```diff
@@ -150,15 +156,21 @@ public Reason getReason() {
 
     /**
      * The timestamp in milliseconds since epoch. Note, we use timestamp here since
-     * we want to make sure its preserved across node serializations. Extra care need
-     * to be made if its used to calculate diff (handle negative values) in case of
-     * time drift.
+     * we want to make sure its preserved across node serializations.
      */
     public long getTimestampInMillis() {
         return this.timestamp;
     }
 
     /**
+     * The timestamp in nanoseconds. Used to calculate the delay for delayed shard allocation.
+     * ONLY EXPOSED FOR TESTS!
+     */
+    public long getTimestampNanos() {
```

Comment:
I think this should have `delay` somewhere in the name, it's kind of confusing to have two `getTimestamp*` methods